cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -323,4 +341,85 @@ private WorkflowToken getWorkflowToken(String namespaceId, String appName, Strin
     }
     return store.getWorkflowToken(workflowId, runId);
   }
+
+  @GET
+  @Path("/apps/{app-id}/workflows/{workflow-id}/runs/{run-id}/localdatasets")
+  public void getWorkflowLocalDatasets(HttpRequest request, HttpResponder responder,
+                                       @PathParam("namespace-id") String namespaceId,
+                                       @PathParam("app-id") String applicationId,
+                                       @PathParam("workflow-id") String workflowId,
+                                       @PathParam("run-id") String runId)
+    throws NotFoundException, DatasetManagementException {
+    WorkflowSpecification workflowSpec = getWorkflowSpecForValidRun(namespaceId, applicationId, workflowId, runId);
+    Map<String, DatasetCreationSpec> localDatasets = new HashMap<>();
+    for (Map.Entry<String, DatasetCreationSpec> localDatasetEntry : workflowSpec.getLocalDatasetSpecs().entrySet()) {
+      String mappedDatasetName = localDatasetEntry.getKey() + "." + runId;
+      if (datasetFramework.hasInstance(Id.DatasetInstance.from(namespaceId, mappedDatasetName))) {
+        localDatasets.put(mappedDatasetName, localDatasetEntry.getValue());
+      }
+    }
+
+    responder.sendJson(HttpResponseStatus.OK, localDatasets);
+  }
+
+  @DELETE
+  @Path("/apps/{app-id}/workflows/{workflow-id}/runs/{run-id}/localdatasets")
+  public void deleteWorkflowLocalDatasets(HttpRequest request, HttpResponder responder,
+                                       @PathParam("namespace-id") String namespaceId,
+                                       @PathParam("app-id") String applicationId,
+                                       @PathParam("workflow-id") String workflowId,
+                                       @PathParam("run-id") String runId) throws NotFoundException {
+    WorkflowSpecification workflowSpec = getWorkflowSpecForValidRun(namespaceId, applicationId, workflowId, runId);
+    Set<String> errorOnDelete = new HashSet<>();
+    for (Map.Entry<String, DatasetCreationSpec> localDatasetEntry : workflowSpec.getLocalDatasetSpecs().entrySet()) {
```

Comment:
Do you think it also makes sense to be able to delete an individual local dataset?